docs/how-to/hip_runtime_api/memory_management/unified_memory.rst

```diff
@@ -0,0 +1,737 @@
+                int *d_a, *d_b, *d_c;
+
+                // Setup input values.
+                a = 1;
+                b = 2;
+
+                // Allocate device copies of a, b and c.
+                HIP_CHECK(hipMalloc(&d_a, sizeof(*d_a)));
+                HIP_CHECK(hipMalloc(&d_b, sizeof(*d_b)));
+                HIP_CHECK(hipMalloc(&d_c, sizeof(*d_c)));
+
+                // Copy input values to device.
+                HIP_CHECK(hipMemcpy(d_a, &a, sizeof(*d_a), hipMemcpyHostToDevice));
+                HIP_CHECK(hipMemcpy(d_b, &b, sizeof(*d_b), hipMemcpyHostToDevice));
+
+                // Launch add() kernel on GPU.
+                hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, d_a, d_b, d_c);
+
+                // Copy the result back to the host.
+                HIP_CHECK(hipMemcpy(&c, d_c, sizeof(*d_c), hipMemcpyDeviceToHost));
+
+                // Cleanup allocated memory.
+                HIP_CHECK(hipFree(d_a));
+                HIP_CHECK(hipFree(d_b));
+                HIP_CHECK(hipFree(d_c));
+
+                // Prints the result.
+                std::cout << a << " + " << b << " = " << c << std::endl;
+
+                return 0;
+            }
+
+.. _using unified memory:
+
+Using unified memory
+================================================================================
+
+Unified memory can simplify the complexities of memory management in GPU
+computing, by not requiring explicit copies between the host and the devices. It
+can be particularly useful in use cases with sparse memory accesses from both
+the CPU and the GPU, as not the whole memory region needs to be transferred to
+the corresponding processor, thereby reducing the amount of memory sent over the
+PCIe bus.
+
+In HIP, pinned memory allocations are coherent by default. Pinned memory is
+host memory mapped into the address space of all GPUs, meaning that the pointer
+can be used on both host and device. Addtionally, using pinned memory instead of
+pageable memory on the host can improve bandwidth for transfers between the host
+and the GPUs.
+
+While unified memory can provide numerous benefits, it's important to be aware
+of the potential performance overhead associated with unified memory. You must
+thoroughly test and profile your code to ensure it's the most suitable choice
+for your use case.
+
+.. _unified memory runtime hints:
+
+Performance optimizations for unified memory
+================================================================================
+
+There are several ways, in which the developer can guide the runtime to reduce
+copies between devices, in order to improve performance.
+
+Data prefetching
+--------------------------------------------------------------------------------
+
+Data prefetching is a technique used to improve the performance of your
+application by moving data to the desired device before it's actually
+needed. ``hipCpuDeviceId`` is a special constant to specify the CPU as target.
```

Comment:
This paragraph expresses that hipCpuDeviceId is a constant to specify the CPU as a target, but is it defined by the runtime? It does not seem to be established in the code example that follows?